### PR TITLE
feat: upgrade various dependencies

### PR DESCRIPTION
**Pull request by xareal**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Android/blob/master/SECURITY.md).
- [] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Android/issues?q=is%3Aissue).

### Issue Description

Newer versions were available.

Related issue: #n/a

### Approach

I just updated the dependencies. 

- Kotlin, 
- facebook login, 
- firebase messaging, 
- android.gms:play-services-auth, 
- fragments, 
- activity-ktx, 
- annotation, 
- okhttp, 
- appcompact
 

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

File: facebook/build.gradle [hunk 1]
```diff
@@ -35,7 +35,7 @@ android {
 }
 
 dependencies {
-    api "com.facebook.android:facebook-login:[13.2.0, 14.0["
+    api "com.facebook.android:facebook-login:13.2.0"
```
Comment (rommansabbir) [inline facebook/build.gradle:38]: Revert back to as it is. Set a range of _version_ instead of a fixed version at this time.

**Review by rommansabbir — CHANGES_REQUESTED** [review]

**Review by mtrezza — CHANGES_REQUESTED** [review]
Just the FB range, otherwise looks good

**Comment by mtrezza** [comment]
> The current version of the Facebook SDK for Android is version 14.1.0 and requires the Android API 15. 

If you want to also upgrade the FB SDK, then I'd do that in a separate PR. As we've seen in the past, since it's a transitive dependency it may require some more deliberation.

File: facebook/build.gradle (hunk 1 above)
Comment (mtrezza) [inline facebook/build.gradle:38]:
```suggestion
    api "com.facebook.android:facebook-login:[13.2.0, 14.0["
```

**Review by mtrezza — APPROVED** [review]
Looks good!
I fixed the commit message (PR title) and reverted the FB lib range.
Is this ready for merge?

File: facebook/build.gradle (hunk 1 above)
Comment (mtrezza) [reply facebook/build.gradle]: Done in https://github.com/parse-community/Parse-SDK-Android/pull/1181/commits/fd1d6027334129529b937e704c32e4861239e09a

**Review by rommansabbir — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.1.0](https://github.com/parse-community/Parse-SDK-Android/releases/tag/4.1.0)

**Comment by xareal** [comment]
Is this a big deal? I feel like this is a big deal for me. I am grateful to
be part of such a big community.

**Pull request merged**